resource/word-processor-plugin-installer.js

```diff
@@ -63,52 +62,32 @@ var ZoteroPluginInstaller = function(addon, failSilently, force) {
 
 	this.prefPaneDoc = null;
 	
-	var me = this;
-	var extensionIDs = [this._addon.EXTENSION_ID].concat(this._addon.REQUIRED_ADDONS.map(req => req.id));
-	Zotero.debug("PluginInstaller: fetching addon info");
-	AddonManager.getAddonsByIDs(extensionIDs, function(addons) {
-		Zotero.debug("PluginInstaller: addon info fetched");
-		me._addonInfo = addons[0];
-		me._addonInfoAvailable();
-	});
+	this.init();
 };
 
 ZoteroPluginInstaller.prototype = {
-	_errorDisplayed: false,
-	
-	_addonInfoAvailable: function() {
+	init: async function() {
+		if (this._initialized) return;
+		Zotero.debug("PluginInstaller: fetching addon info");
+		Zotero.debug("PluginInstaller: addon info fetched");
+		
 		try {
-			this._version = this._addonInfo.version;
-			
-			try {
-				this._addon.verifyNotCorrupt(this);
-			} catch(e) {
-				Zotero.debug("Not installing +this._addon.EXTENSION_STRING+:  "+e.toString());
-				return;
-			}
-			
-			var version = this.prefBranch.getCharPref("version");			
-			if(this.force || (
-					(
-						Services.vc.compare(version, this._addon.LAST_INSTALLED_FILE_UPDATE) < 0
-						|| (!Zotero.isStandalone && !this.prefBranch.getBoolPref("installed"))
-					)
-					&& !this.prefBranch.getBoolPref("skipInstallation")
-				)) {
-					
-				var me = this;
+			this._version = await Zotero.File.getContentsFromURLAsync(this._addon.VERSION_FILE);
```

Comment:
Can we add a `trim()` here just to avoid some future breakage if we edit in the wrong text editor? Not sure what `setCharPref()` would do with a trailing newline.